contractcourt/channel_arbitrator.go

```diff
@@ -1132,7 +1137,29 @@ func (c *ChannelArbitrator) shouldGoOnChain(htlcExpiry, broadcastDelta,
 
 	// We should on-chain for this HTLC, iff we're within out broadcast
 	// cutoff window.
-	return currentHeight >= broadcastCutOff
+	if currentHeight < broadcastCutOff {
+		return false
+	}
+
+	// In case of incoming htlc we should go to chain.
+	if htlc.Incoming {
+		return true
+	}
+
+	// For htlcs that are result of our initiated payments we give some grace
+	// period before force closing the channel. During this time we expect
+	// both nodes to connect and give a chance to the other node to send its
+	// updates and cancel the htlc.
+	// This shouldn't add any security risk as there is no incoming htlc to
+	// fulfill at this case and the expectation is that when the channel is
+	// active the other node will send update_fail_htlc to remove the htlc
+	// without closing the channel. It is up to the user to force close the
+	// channel if the peer misbehaves and doesn't send the update_fail_htlc.
+	// It is useful when this node is most of the time not online and is
+	// likely to miss the time slot where the htlc may be cancelled.
+	hasForward := c.cfg.HasForwardedHTLCWithHash(htlc.RHash)
```

Comment:
We know the channel id here and the htlc index, so it could be a lookup by outgoing circuit instead of looking up all circuits with that hash.

Reply:
I see now you probably mean to use `LookupOpenCircuit` to fetch the exact circuit.
To execute this query I need to build the `CirtuitKey`. Quick question: Is the `HtlcIndex` of the `channeldb.HTLC` represents the `HtlcID` of the `CirtuitKey` ?

Reply:
I pushed changes with this feedback.

Reply:
Ok, looks good. Yes, `HtlcIndex` is `HtlcId`.